Code/Framework/AzCore/Platform/Common/UnixLike/AzCore/Debug/Trace_UnixLike.cpp

```diff
@@ -6,74 +6,123 @@
  *
  */
 
+#include <AzCore/Debug/TraceMessageBus.h>
 #include <AzCore/IO/SystemFile.h>
 #include <AzCore/std/string/string_view.h>
+#include <AzCore/Debug/StackTracer.h>
 
-#include <ctype.h>
 #include <signal.h>
-#include <unistd.h>
 
-namespace AZ::Debug::Platform
+namespace AZ::Debug
 {
 #if defined(AZ_ENABLE_DEBUG_TOOLS)
-    bool performDebuggerDetection()
+    void ExceptionHandler(int signal);
+#endif
+
+    constexpr int MaxMessageLength = 4096;
+    constexpr int MaxStackLines = 100;
+
+    namespace Platform
     {
-        AZ::IO::SystemFile processStatusFile;
-        if (!processStatusFile.Open("/proc/self/status", AZ::IO::SystemFile::SF_OPEN_READ_ONLY))
+#if defined(AZ_ENABLE_DEBUG_TOOLS)
+        bool performDebuggerDetection()
         {
+            AZ::IO::SystemFile processStatusFile;
+            if (!processStatusFile.Open("/proc/self/status", AZ::IO::SystemFile::SF_OPEN_READ_ONLY))
+            {
+                return false;
+            }
+
+            char buffer[4096];
+            AZ::IO::SystemFile::SizeType numRead = processStatusFile.Read(sizeof(buffer), buffer);
+
+            const AZStd::string_view processStatusView(buffer, buffer + numRead);
+            constexpr AZStd::string_view tracerPidString = "TracerPid:";
+            const size_t tracerPidOffset = processStatusView.find(tracerPidString);
+            if (tracerPidOffset == AZStd::string_view::npos)
+            {
+                return false;
+            }
+            for (size_t i = tracerPidOffset + tracerPidString.length(); i < numRead; ++i)
+            {
+                if (!::isspace(processStatusView[i]))
+                {
+                    return processStatusView[i] != '0';
+                }
+            }
             return false;
         }
 
-        char buffer[4096];
-        AZ::IO::SystemFile::SizeType numRead = processStatusFile.Read(sizeof(buffer), buffer);
+        bool IsDebuggerPresent()
+        {
+            static bool s_detectionPerformed = false;
+            static bool s_debuggerDetected = false;
+            if (!s_detectionPerformed)
+            {
+                s_debuggerDetected = performDebuggerDetection();
+                s_detectionPerformed = true;
+            }
+            return s_debuggerDetected;
+        }
 
-        const AZStd::string_view processStatusView(buffer, buffer + numRead);
-        constexpr AZStd::string_view tracerPidString = "TracerPid:";
-        const size_t tracerPidOffset = processStatusView.find(tracerPidString);
-        if (tracerPidOffset == AZStd::string_view::npos)
+        bool AttachDebugger()
         {
+            // Not supported yet
+            AZ_Assert(false, "AttachDebugger() is not supported for Unix platform yet");
             return false;
         }
-        for (size_t i = tracerPidOffset + tracerPidString.length(); i < numRead; ++i)
+
+        void SignalHandler(int handler)
+        {
+        }
```

Comment:
This function is unused.
It looks like ExceptionHandler is what is being used to handle the signal instead.